src/DevChatter.DevStreams.Web/Controllers/IsLiveController.cs

```diff
@@ -26,20 +27,47 @@ public class IsLiveController : Controller
 
 
         /// <summary>
-        /// Get all live streams.
+        /// Get all live streams info.
         /// </summary>
-        /// <returns></returns>
+        /// <returns>TwitchName, isLive = True, started_At, View Count</returns>
         [HttpGet]
         public async Task<IActionResult> Get()
         {
-            List<TwitchChannel> channels = await _crudRepository.GetAll<TwitchChannel>();
-            List<string> twitchIds = channels.Select(x => x.TwitchId).ToList();
-            var liveTwitchIds = (await _twitchService.GetChannelLiveStates(twitchIds))
+            List<Channel> channels = await _channelAggregateService.GetAllAggregates();
+            List<string> twitchIds = channels.Select(x => x?.Twitch?.TwitchId).ToList();
+            twitchIds.RemoveAll(string.IsNullOrWhiteSpace);
+            var liveTwitchData = (await _twitchService.GetChannelLiveStates(twitchIds));
+
+            var sortedLiveData = liveTwitchData.OrderByDescending(o => o.ViewerCount).ToList();
+
+            var liveTwitchIds = sortedLiveData
                 .Where(x => x.IsLive)
                 .Select(x => x.TwitchId)
                 .ToList();
-            var liveChannelNames = channels.Where(c => liveTwitchIds.Contains(c.TwitchId)).Select(c => c.TwitchName);
-            return Ok(liveChannelNames);
+
+            var liveChannelSorted = sortedLiveData
+                .OrderByDescending(x => x.ViewerCount)
+                .Select(x => channels.Where(y => y?.Twitch?.TwitchId == x?.TwitchId).Where(v => x.IsLive).Select(z => z?.Twitch?.TwitchName))
+                .Where(x => x.Any())
+                .ToList();
+
+            var timeDifference = sortedLiveData
+                .Where(x => liveTwitchIds.Contains(x.TwitchId))
+                .Select(x => (DateTime.UtcNow - x?.StartedAt.ToUniversalTime()));
+
+            var viewerCount = sortedLiveData
+                .Where(x => liveTwitchIds.Contains(x.TwitchId))
+                .Select(x => x.ViewerCount)
+                .ToList();
+
+            var responseObject = new
```

Comment:
Why aren't we just returning back a variation on the LiveState objects, which have the properties for Channel, ViewCount, TimeOnline?